config/locales/blacklight.en.yml

```diff
@@ -140,7 +140,7 @@ en:
 
     search:
       title: '%{application_name} Search Results'
-      search_results_header: 'Search'
+      search_results_header: 'Search Constraints'
```

Comment:
I'm not sure about the new translation, although, I think you're right that the original translation wasn't very good. Here's the context for anyone with better ideas:

https://github.com/projectblacklight/blacklight/blob/50acc870be1a81352255fdeb4d75d92ac7eeb67a/app/views/catalog/_search_results.html.erb#L1